sale_order_move_menu/views/stock_move_views.xml

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="stock_move_move_menu_search" model="ir.ui.view">
+        <field name="name">stock.move.search (Sale Moves)</field>
+        <field name="model">stock.move</field>
+        <field name="arch" type="xml">
+            <search string="Stock Moves">
+                <field name="sale_order" string="Sale Order"/>
+                <field name="product_id"/>
+                <field name="product_categ_id" />
+                <separator />
+                <filter string="Today" name="today" domain="[('date','&gt;', (context_today() - datetime.timedelta(days=1)).strftime('%%Y-%%m-%%d'))]"/>
+                <filter string="Last Day" name="last_day" domain="[('date','&gt;', (context_today() - datetime.timedelta(days=2)).strftime('%%Y-%%m-%%d'))]"/>
+                <filter string="Last Week" name="last_week" domain="[('date','&gt;', (context_today() - datetime.timedelta(days=7)).strftime('%%Y-%%m-%%d'))]"/>
+                <filter string="Last Month" name="last_month" domain="[('date','&gt;', (context_today() - datetime.timedelta(days=30)).strftime('%%Y-%%m-%%d'))]"/>
+                <filter string="Date" name="filter_date" date="date"/>
```

Comment:
Default period is possible to be set, see https://www.odoo.com/documentation/13.0/developer/reference/addons/views.html , at this section:

![imagen](https://user-images.githubusercontent.com/12749832/149380038-fdd63544-7885-461c-8972-dc653d719c48.png)

When set, combined with action filter context, it's possible to access the menu with other filter than "this month". By the way, you can only check, for me "this month" is a good initial filter